packages/fx-core/src/core/environment.ts

```diff
@@ -385,6 +392,50 @@ class EnvironmentManager {
       return this.defaultEnvName;
     }
   }
+
+  public getActiveEnv(projectRoot: string): Result<string, FxError> {
+    if (!isMultiEnvEnabled()) {
+      return ok("default");
+    }
+    if (!isValidProject(projectRoot)) {
+      return err(InvalidProjectError());
+    }
+    const settingsJsonPath = path.join(
+      projectRoot,
+      `.${ConfigFolderName}/${InputConfigsFolderName}/${ProjectSettingsFileName}`
+    );
+    let settingsContent;
+    try {
+      settingsContent = fs.readFileSync(settingsJsonPath, "utf8");
+    } catch (error) {
+      return err(ReadFileError(error));
+    }
+
+    let settingsJson;
+    try {
+      settingsJson = JSON.parse(settingsContent);
+    } catch (error) {
+      return err(
+        InvalidProjectSettingsFileError(
+          `Project settings file is not a valid JSON, error: '${error}'`
+        )
+      );
+    }
```

Comment:
nit: Here we can use `fs.readJSON` for simplicity.

Reply:
It makes sense. Fixed